ably/realtime_conn.go

```diff
@@ -31,14 +32,21 @@ const (
 // Connection represents a single connection Realtime instantiates for
 // communication with Ably servers.
 type Connection struct {
+	mtx sync.Mutex
+
+	ConnectionEventEmitter
+
+	state           ConnectionState
+	errorReason     *ErrorInfo
+	internalEmitter ConnectionEventEmitter
```

Comment:
redundant? We should just use the embedded `ConnectionEventEmitter` so we can have a single emitter per `Connection`.

Reply:
See https://github.com/ably/ably-go/pull/190#discussion_r478996084.